subsys/bluetooth/services/fast_pair/fp_advertising.c

```diff
@@ -20,17 +21,25 @@
 #define FIELD_LEN_TYPE_SIZE			sizeof(uint8_t)
 #define ENCODE_FIELD_LEN_TYPE(len, type)	(((len) << TYPE_BITS) | (type))
 
+#define BATTERY_LEVEL_BITS			7
+#define BATTERY_VALUE_SIZE			sizeof(uint8_t)
+#define ENCODE_FIELD_BATTERY_STATUS_LEVEL(charging, level) \
+						(((charging) << BATTERY_LEVEL_BITS) | (level))
```

Comment:
Make sure that charging bit is either `1` or `0`. In C anything but not zero is true.